builtin/providers/aws/autoscaling_tags.go

```diff
@@ -115,31 +137,83 @@ func diffAutoscalingTags(oldTags, newTags []*autoscaling.Tag, resourceID string)
 	return autoscalingTagsFromMap(create, resourceID), remove
 }
 
+func autoscalingTagsFromList(vs []interface{}, resourceID string) []*autoscaling.Tag {
+	result := make([]*autoscaling.Tag, 0, len(vs))
+	for _, tag := range vs {
+		attr, ok := tag.(map[string]interface{})
+		if !ok {
+			continue
+		}
+
+		if t := autoscalingTagFromMap(attr, resourceID); t != nil {
+			result = append(result, t)
+		}
+	}
+	return result
+}
+
 // tagsFromMap returns the tags for the given map of data.
 func autoscalingTagsFromMap(m map[string]interface{}, resourceID string) []*autoscaling.Tag {
 	result := make([]*autoscaling.Tag, 0, len(m))
-	for k, v := range m {
-		attr := v.(map[string]interface{})
-		t := &autoscaling.Tag{
-			Key:               aws.String(k),
-			Value:             aws.String(attr["value"].(string)),
-			PropagateAtLaunch: aws.Bool(attr["propagate_at_launch"].(bool)),
-			ResourceId:        aws.String(resourceID),
-			ResourceType:      aws.String("auto-scaling-group"),
+	for _, v := range m {
+		attr, ok := v.(map[string]interface{})
+		if !ok {
+			continue
 		}
-		if !tagIgnoredAutoscaling(t) {
+
+		t := autoscalingTagFromMap(attr, resourceID)
+		if t != nil {
 			result = append(result, t)
 		}
 	}
 
 	return result
 }
 
+func autoscalingTagFromMap(attr map[string]interface{}, resourceID string) *autoscaling.Tag {
+	if _, ok := attr["key"]; !ok {
+		return nil
+	}
+
+	if _, ok := attr["value"]; !ok {
+		return nil
```

Comment:
Is there any reason to ignore such case as it's an invalid block of code? I think it would make more sense to just return error in case that `key`, `value` or `propagate_at_launch` are not defined.